### DISCUSSION TITLE
Heltec ESP32 LoRa v3, the unofficial library

### DISCUSSION THREAD

**Discussion by ropg** [jgromes/RadioLib#1007] [Show and tell]
Hi all,

I created an [unofficial library for the Heltec ESP32 LoRa v3 development board](https://github.com/ropg/heltec_esp32_lora_v3). It's such cool hardware, and it's just way too complicated and cumbersome to play with it using the official software they provide. I tried to make all the features of the board accessible and document everything really carefully. Naturally the library uses RadioLib and not whatever the stack was that they suggest you use. Hope you like it.

[<img width="840" alt="image" src="https://github.com/jgromes/RadioLib/assets/343196/5efb4bdb-0b24-47aa-a9db-ddb2d69714b4">](https://github.com/ropg/heltec_esp32_lora_v3)

**Comment by ropg** [comment]
The [**library**](https://github.com/ropg/heltec_esp32_lora_v3) now also has Heltec Wireless Stick v3 support

<img width="295" alt="image" src="https://github.com/jgromes/RadioLib/assets/343196/a244a3cd-eb77-4a67-b679-790f6d89763d">

**Reply by Telegiangi61** [reply]
hello Ropg, first of all thanks for your work, i have a question about wireless stick V3 join to Lora OpenMqtt, they not communicate i for me, i suppose due lack of syncword parameter on your library for private network (0x12). may be i loss some documentation, but could you explain how set this parameter on your library? many thanks

**Reply by ropg** [reply]
Not sure what the problem is: "they not communicate" is not much to go by. I have no experience with LoRa OpenMqtt beyond having just Googled it. This library merely provides a well-documented interface to use all the functionality in the Heltec v3 boards. As for RadioLib: it simply provides the usual `radio` instance as well as a few convenience functions to call RadioLib and deal with return codes.

I did write another library to manage LoRaWAN session data as well as LoRaWAN provisioning data but that is also not what you're looking for.

**Reply by tao480** [reply]
I'm using this Arduino library for MQTT and it works fine:
#include <PubSubClient.h>  //A simple client for MQTT.

**Comment by HeadBoffin** [comment]
Closing so this doesn't become a support thread for someone else's repro.
